In [1]:
import pandas as pd
import statsmodels.api as sm
import scipy.stats as ss
import math
import statistics
from sklearn.metrics import mean_squared_error

<h2>1) Using the posted data set ECONMATH2.DTA to answer the following questions.</h2>

In [4]:
econmath_df = pd.read_stata('ECONMATH2.dta')
econmath_df.dropna()
econmath_df.head(5)

colgpa  hsgpa  actmth  acteng  study
0  3.4909  3.355      26      24   10.0
1  2.1000  3.219      20      23   22.5
2  3.0851  3.306      24      21   12.0
3  2.6805  3.977      28      31   40.0
4  3.7454  3.890      31      28   15.0

In [5]:
x = econmath_df[['hsgpa', 'actmth', 'acteng']]
y = econmath_df[['colgpa']]
x = sm.add_constant(x)
reg = sm.OLS(y,x).fit()
print(reg.summary())

                            OLS Regression Results                            
Dep. Variable:                 colgpa   R-squared:                       0.256
Model:                            OLS   Adj. R-squared:                  0.253
Method:                 Least Squares   F-statistic:                     92.76
Date:                Wed, 08 Sep 2021   Prob (F-statistic):           1.31e-51
Time:                        23:47:14   Log-Likelihood:                -531.52
No. Observations:                 814   AIC:                             1071.
Df Residuals:                     810   BIC:                             1090.
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.0282      0.168      0.168      0.8

<h3>(a) Estimate a model relating colgpa (dependent variable) to hsgpa, actmth, and acteng (indepen-
dent variables). Report the results using the usual Stata output. Are all explanatory variables
signi…cant at conventional levels? Discuss.</h3>

<p>𝑐𝑜𝑙𝑔𝑝𝑎̂ = 𝛽<sub>0</sub> + 𝛽<sub>1</sub>ℎ𝑠𝑔𝑝𝑎 + 𝛽<sub>2</sub>𝑎𝑐𝑡𝑚𝑡ℎ + 𝛽<sub>3</sub>𝑎𝑐𝑡𝑒𝑛𝑔
<br>
&emsp; =0.0282 + 0.6590 ℎ𝑠𝑔𝑝𝑎 + 0.0130 𝑎𝑐𝑡𝑚𝑡ℎ + 0.0122 𝑎𝑐𝑡𝑒𝑛𝑔</p>
<p>By comparing the P-value with 𝛼, hsgpa has the P-value of 0.000, actmth has the P-value of 0.012, and acteng has the P-value of 0.016, which implies that hsgpa is significant at 1%, 5%, and 10% level, while actmth and acteng is significant at 5% and 10% level since actmth and acteng has P-value higher than 0.01 but lower than 0.05 and 0.10.</p>

<h3>(b) Consider an increase in hsgpa of one standard deviation (about 0.343). By how much does colgpa
increase, holding actmth and acteng fixed. About how many standard deviations would the
actmth have to increase to change colgpa by the same amount as a one standard deviation in
hsgpa? Explain.</h3>

<p>Standard deviation for hsgpa is about 0.343, the coefficient is .6590007. So an increase of one standard deviation on hsgpa would increase colgpa by</p>
<p>Δ𝑐𝑜𝑙𝑔𝑝𝑎̂ = 𝛽̂ℎ𝑠𝑔𝑝𝑎 ∗ 𝑠𝑑(ℎ𝑠𝑔𝑝𝑎) = (0.6590007)(0.343) ≈ 0.2260372401</p>
<p>As for how many standard deviations would the actmth have to increase to change 𝑐𝑜𝑙𝑔𝑝𝑎̂ by the same amount as a one standard deviation in hsgpa, it would be</p>
<p>Δ𝑠𝑑_𝑎𝑐𝑡𝑚𝑡ℎ = 𝛽̂<sub>ℎ𝑠𝑔𝑝𝑎</sub> ∗ 𝑠t𝑑(ℎ𝑠𝑔𝑝𝑎) / 𝛽̂<sub>𝑎𝑐𝑡𝑚𝑡ℎ</sub> ∗ 𝑠t𝑑(𝑎𝑐𝑡𝑚𝑡ℎ) = (0.6590007)(0.343)(0.0130198)(3.773) ≈ 4.606</p>

<h3>(c) Test the null hypothesis that actmth and hsgpa have the same e¤ect (in the population) against
a two-sided alternative. Report the p-value and explain your conclusion</h3>

In [6]:
# ss.ttest_rel(econmath_df['actmth'], econmath_df['hsgpa'])
hypotheses = '(actmth=0), (hsgpa=0)'
f_test = reg.f_test(hypotheses)
print(f_test)

<F test: F=array([[96.23609624]]), p=3.1778773828316035e-38, df_denom=810, df_num=2>


<p>𝐻<sub>0</sub>: 𝛽𝑎𝑐𝑡𝑚𝑡ℎ = 𝛽ℎ𝑠𝑔𝑝𝑎
    <br>𝐻<sub>1</sub>: 𝛽𝑎𝑐𝑡𝑚𝑡ℎ ≠ 𝛽ℎ𝑠𝑔𝑝𝑎</p>
<p>Could be rewritten as:</p>
<p>𝐻<sub>0</sub>: 𝛽𝑎𝑐𝑡𝑚𝑡ℎ − 𝛽ℎ𝑠𝑔𝑝𝑎 = 0
    <br>𝐻<sub>1</sub>: 𝛽𝑎𝑐𝑡𝑚𝑡ℎ − 𝛽ℎ𝑠𝑔𝑝𝑎 ≠ 0</p>

<p>the p-value of 0.000 was reported back. This means we reject the null hypothesis, which implies that the probability of the null hypothesis being true is ≈ 0.000</p>

<h3>(d) Now include the variable study into the model. Using the formula for the F-statistic, test the
null that impact of actmth, acteng, and study are all jointly equal to zero. Show your work and
report the calculated F-statistic and p-value.</h3>

In [7]:
x = econmath_df[['hsgpa', 'actmth', 'acteng', 'study']]
x = sm.add_constant(x)
reg = sm.OLS(y,x).fit()
print(reg.summary())

                            OLS Regression Results                            
Dep. Variable:                 colgpa   R-squared:                       0.258
Model:                            OLS   Adj. R-squared:                  0.254
Method:                 Least Squares   F-statistic:                     70.38
Date:                Thu, 09 Sep 2021   Prob (F-statistic):           3.69e-51
Time:                        00:02:08   Log-Likelihood:                -530.19
No. Observations:                 814   AIC:                             1070.
Df Residuals:                     809   BIC:                             1094.
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         -0.0128      0.169     -0.075      0.9

In [8]:
hypotheses = '(actmth=0), (acteng=0), (study=0)'
f_test = reg.f_test(hypotheses)
print(f_test)

<F test: F=array([[7.70572089]]), p=4.429992896085183e-05, df_denom=809, df_num=3>


In [16]:
print('The F critical value at 1% is')
print(ss.f.ppf(1-0.01,3,809))
print('The F critical value at 5% is')
print(ss.f.ppf(1-0.05,3,809))
print('MSE is ')
print(reg.mse_model)

The F critical value at 1% is
3.8058973062495607
The F critical value at 5% is
2.615908847783631
MSE is 
3.905615457458832


<p>Since F statistics is approximately 7.71, which is way above F critical value at 1% significance level, we reject the null hypothesis</p>

<p>Also, the p value is approximately 0.000043, which is way below the alpha, we reject the null hypothesis with p value method as well, which corresponds with F test result</p>

<h2>2) Using the data in SLEEP75.DTA, estimate the following]
<br>
<br>
&emsp;    sleep = 𝛽<sub>0</sub> + 𝛽<sub>1</sub>totwrk + 𝛽<sub>2</sub>educ + 𝛽<sub>3</sub>age + u
<br>
<br>
where sleep is minutes of sleep per week, totwrk is minutes of work per week, educ is years of schooling,
and age is age in years.</h2>

In [15]:
sleep75_df = pd.read_stata('SLEEP75.dta')
sleep75_df.dropna(subset=['totwrk','educ','age'])
sleep75_df.head(5)

age  black  case  clerical  construc  educ  earns74  gdhlth  inlf  leis1  \
0   32      0     1       0.0       0.0    12      0.0       0     1   3529   
1   31      0     2       0.0       0.0    14   9500.0       1     1   2140   
2   44      0     3       0.0       0.0    17  42500.0       1     1   4595   
3   30      0     4       0.0       0.0    12  42500.0       1     1   3211   
4   64      0     5       0.0       0.0    14   2500.0       1     1   4052   

   ...  spwrk75  totwrk  union  worknrm  workscnd  exper  yngkid  yrsmarr  \
0  ...        0    3438      0     3438         0     14       0       13   
1  ...        0    5020      0     5020         0     11       0        0   
2  ...        1    2815      0     2815         0     21       0        0   
3  ...        1    3786      0     3786         0     12       0       12   
4  ...        1    2580      0     2580         0     44       0       33   

      hrwage  agesq  
0   7.070004   1024  
1   1.429999    961  
2  20.529997   1936  
3   9.619998    900  
4   2.750000   4096  

[5 rows x 34 columns]

In [16]:
x = sleep75_df[['totwrk', 'educ', 'age']]
y = sleep75_df[['sleep']]
x = sm.add_constant(x)
reg = sm.OLS(y,x).fit()
print(reg.summary())

                            OLS Regression Results                            
Dep. Variable:                  sleep   R-squared:                       0.113
Model:                            OLS   Adj. R-squared:                  0.110
Method:                 Least Squares   F-statistic:                     29.92
Date:                Thu, 09 Sep 2021   Prob (F-statistic):           3.28e-18
Time:                        00:22:49   Log-Likelihood:                -5263.1
No. Observations:                 706   AIC:                         1.053e+04
Df Residuals:                     702   BIC:                         1.055e+04
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const       3638.2453    112.275     32.405      0.0

<h3>(a) Is either age or educ individually signi…cant at the 5% level against a two-sided alternative with
the hypothesized value equal to zero? Explain and show your work.</h3>

In [17]:
ss.t.ppf(q=1-.05/2,df=702) # t critical value

1.9633490231077653

<p>the t critical value is approximately 1.96.</p>
<p>For educ, the t-statistics is -1.89, which makes |tc| > |tB|, and the p-value of 0.059, which is higher than 0.05, imply that we fail to reject the null hypothesis for educ by both t-statistics method and p-value method at the significance level of 5%.</p>
<p>As for age, the t-statistics is 1.52, which makes |tc| > |tB|, and the p-value of 0.129, which is higher than 0.05, imply that we fail to reject the null hypothesis for age by both
t-statistics method and p-value method at the significance level of 5%.</p>

<h3>(b) Are educ and age jointly signi…cant at the 5% level? Compute the F-statistic using the formula,
and then state clearly whether you would reject or fail to reject the null.</h3>

In [18]:
hypotheses = '(educ=0), (age=0)'
f_test = reg.f_test(hypotheses)
print(f_test)

<F test: F=array([[3.9890998]]), p=0.018937671866617236, df_denom=702, df_num=2>


In [19]:
print('The F critical value is')
print(ss.f.ppf(1-0.05,2,702))

The F critical value is
3.0085527837787773


<p>Since F statistics value is approximately 3.99, which is higher than F critical value, which is approximately 3.009, we reject the null hypothesis by F test method</p>

<p>Also, the p value is approximately 0.0189, which is a bit higher than 0.01 but way below 0.05, we fail to reject the null hypothesis at 1% significance level but reject the null hypothesis at 5% significance level</p>